images/cassandra/tests/main.tf

```diff
@@ -0,0 +1,181 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "~> 2.11.0"
+    }
+  }
+}
+
+provider "kubernetes" {
+  config_path    = "~/.kube/config"
+  config_context = "kind-kind"
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "oci_exec_test" "test" {
+  for_each = toset(fileset(path.module, "*.sh"))
+
+  digest = var.digest
+  script = "${path.module}/${each.value}"
+}
+
+resource "kubernetes_manifest" "prerequisites" {
+  for_each = fileset(path.module, "manifests/*.yaml")
+  manifest = yamldecode(file("${path.module}/${each.value}"))
+}
+
+# resource "kubectl_manifest" "stateful_set" {
+#   depends_on = [kubernetes_manifest.prerequisites]
+#   yaml_body = templatefile("${path.module}/manifests/cassandra-stateful-set.yaml.tpl",
+#     {
+#       cassandra_tag  = data.oci_string.ref.pseudo_tag
+#       cassandra_repo = data.oci_string.ref.registry_repo
+#   })
+# }
+
+resource "kubernetes_stateful_set" "cassandra" {
+  depends_on = [kubernetes_manifest.prerequisites]
+  metadata {
+    name      = "cassandra"
+    namespace = "cassandra"
```

Comment:
```suggestion
  metadata {
    name      = "cassandra"
    namespace = kubernetes_namespace.cassandra.metadata[0].name
```

Reply:
same goes for the `kubernetes_service` that also gets created.

I think this should fix your "namespace not found" issue since the signaling between resource completions is better

Reply:
Got it, thanks a lot 🥺